### ISSUE TITLE
`nerdctl start` cannot start containers in **Created** or **Exited** status 

### ISSUE THREAD

**Issue by jls-0** [containerd/nerdctl#492]
**Required behavior:** Allow start Created or Exited containers, either with `nerdctl start` or `nerdctl exec -it`.
**Actual behavior:**
you can run a container, for example: `nerdctl run -it nginx:latest --name nginx-container`.
You can stop it: `nerdctl stop nginx-container`.
**Issue:** 
You cannot start again the container neither `nerdctl start nginx-container` nor `nerdctl exec -it nginx-container /bin/sh`.
**Error messages:**
`nerdctl start nginx-container`:
`FATA[0000] OCI runtime create failed: cannot allocate tty if runc will detach without setting console socket: unknown`
`nerdctl exec -it nginx-container /bin/sh`:
`FATA[0000] no running task found: task 0a88555b90488894c72918665b9b14bd6ae96e89676d65ca772b9abf34727884 not found: not found`
**Environment:**
  - OS: MacOs Big Sur 11.0.1.
  - Backend: Rancher Desktop 0.6.0 (nerdctl 0.12.1).

**Comment by AkihiroSuda** [comment]
-it is currently incomparable with stop/start, please use -d

**Comment by jls-0** [comment]
> -it is currently incomparable with stop/start, please use -d

It didn't work, I created and ran the container with `nerdctl run --name my-container **-d** nginx`, even modified the image to set the ENTRYPOINT to `/bin/bash`; the container exited immediately because there's no interactive nor tty attached, so there's no process to maintain running in background.
Then, I tried to start or exec the exited container but got the following output:
`nerdctl start my-container`: started the container but exited immediately due to  the mentioned above.
`nerdctl exec -it my-container /bin/sh`: `FATA[0000] cannot exec in a stopped state: unknown`, which is logical because we cannot execute command on stopped containers.

**Comment by jls-0** [comment]
I think in the meantime it cannot be achieve until the implementation of the `create` action from docker, because it allows us to define the `-it` parameters at creation time, then we can start the container in background allowing the interactive/tty connection for later.

**Comment by pvbiesen** [comment]
+1 migrating from docker, very annoying `nerdctl start -i <container> /bin/bash` doesn't work.

**Comment by animatedcreativity** [comment]
Yes, cannot start stopped containers.

**Comment by raktim-sixt** [comment]
yes, same here...cannot start the container which is in the `CREATED` state

```failed to create shim task: OCI runtime create failed: runc create failed: cannot allocate tty if runc will detach without setting console socket: unknown```

has anyone found any workaround/solution to this?

**Comment by aerphanas** [comment]
is anyone know how to run image properly ?
i run image with : 

```sh
nerdctl run -it overlayfs@sha256:564f654982832faaeb09ab49b4b543b83a79f478b5d73bc2785fe0aa07648597
```

after i stop container and i want to run it again with : 

```sh
nerdctl container start overlayfs@sha256:564f654982832faaeb09ab49b4b543b83a79f478b5d73bc2785fe0aa07648597
```

it say

```sh
FATA[0000] failed to create shim task: OCI runtime create failed: runc create failed: cannot allocate tty if runc will detach without setting console socket: unknown
```
